airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryRecordConsumer.java

```diff
@@ -96,10 +98,13 @@ public void acceptTracked(AirbyteMessage message) {
             String.format("Message contained record from a stream that was not in the catalog. \ncatalog: %s , \nmessage: %s",
                 Jsons.serialize(catalog), Jsons.serialize(recordMessage)));
       }
+      final BigQueryWriteConfig writer = writeConfigs.get(pair);
       try {
-        final BigQueryWriteConfig writer = writeConfigs.get(pair);
         writer.getWriter().write(ByteBuffer.wrap((Jsons.serialize(formatRecord(writer.getSchema(), recordMessage)) + "\n").getBytes(Charsets.UTF_8)));
-      } catch (IOException e) {
+      } catch (IOException | RetryHelperException e) {
+        LOGGER.error("Got an error while writing message:" + e.getMessage());
```

Comment:
let's log as much as we possibly can here if the status is 404: 

* region
* number of input streams 
* all job information 
* any other info that would be helpful

Reply:
Hi @sherifnada.
Many thanks for your comment.
I've updated the PR with adding more logs. But the point here is that at the moment when we get an exception - we get it all wrapped as a String. So the only way to check if 404 is "string.contains()" method that looks a little bit ugly for this case as for me. But here is the thing, if we fail at some moment for other reason we anyway would like to know all the details and the whole "sync attempt" would be marked as failed. So I added full logs without additional verification for reason. Please let me know if you don't agree. 
By the way, this PR is in "draft" status for now and waiting for GL internal code review.